### ISSUE TITLE
Allow resourcet-1.3 in broadcast-chan-conduit

### ISSUE THREAD

**Issue by jchia** [merijn/broadcast-chan#20]
There is a breaking change in resourcet-1.3 related to `Data.Acquire.ReleaseType` where `ReleaseException` is replaced by `ReleaseExceptionWith`.

This affects `BroadcastChan.Conduit.Internal`, which uses `ReleaseException`, including for constructing a `BroadcastChan.Extra.ThreadBracket` wherein `cleanupForkError` does not get the exception to use for `ReleaseExceptionWith` value construction. Thus, the changes necessary for using resourcet-1.3 are non-trivial and require code changes and not just .cabal changes, and perhaps even `BroadcastChan.Extra.ThreadBracket` needs to change.

**Comment by merijn** [comment]
Yeah, that's been a thorn in my side for a while. I was working on a fix at one point, but that got stuck and ended up on the low priority pile for awhile.

**Comment by jchia** [comment]
Have you decided what designs are acceptable, though? It seems that an interface change in `ThreadBracket` is inevitable in a new version of broadcast-extra that uses resourcet-1.3.
